runtime/runtime/src/lib.rs

```diff
@@ -226,12 +226,14 @@ impl ActionResult {
             next_result.gas_burnt,
             next_result.gas_used
         );
-        self.gas_burnt = safe_add_gas(self.gas_burnt, next_result.gas_burnt)?;
-        self.gas_burnt_for_function_call = safe_add_gas(
-            self.gas_burnt_for_function_call,
-            next_result.gas_burnt_for_function_call,
-        )?;
-        self.gas_used = safe_add_gas(self.gas_used, next_result.gas_used)?;
+        self.gas_burnt =
+            self.gas_burnt.checked_add(next_result.gas_burnt).ok_or(IntegerOverflowError)?;
+        self.gas_burnt_for_function_call = self
+            .gas_burnt_for_function_call
+            .checked_add(next_result.gas_burnt_for_function_call)
+            .ok_or(IntegerOverflowError)?;
```

Comment:
It feels overly verbose to type `ok_or(IntegerOverflowError)` every time. Can we make `checked_add_result()` method on Gas type?

Reply:
We can't do that on Gas type as the error type is in near-primitives and Gas type is in near-primitives-core. Though we can use safe_add_gas:)